### PR TITLE
New component Tooltip

### PR DESCRIPTION
**Pull request by samuelg0rd0n**
New component Tooltip - simple wrapper over [@tippyjs/react](https://github.com/atomiks/tippyjs-react). We are already using this library for Dropdown component.

Also, Icon component was improved by including `forwardRef` so that it can be used as a clickable/hoverable target for Tooltip (or Dropdown).

### CODE REVIEW

File: src/components/Icon/IconStyles.tsx [hunk 1]
```diff
@@ -5,28 +5,34 @@ import { Spacing } from '../../types/Spacing'
...
 
-export const StyledIcon = styled(SVG)<StyledIconProps>`
-	display: inline-block;
-	${({ size, theme }): string => {
-		if (Spacing.includes(size as unknown as Spacing)) {
+export const StyledIconContainer = styled.span<StyledIconProps>`
+	display: inline-flex;
+
+	${({ $size, theme }): string => {
+		if (Spacing.includes($size as unknown as Spacing)) {
```
Comment (chorobin) [inline src/components/Icon/IconStyles.tsx:15]: Would it be cleaner to create a type guard for this?

```
const isSpacing = (size: Spacing | string | number): size is Spacing => {
  return Spacing.includes($size as unknown as Spacing)
}
```

And then using the type guard in a if statement should always infer the correct type

```
if (isSpacing($size)) {
// $size should be Spacing
} else if (typeof $size === 'number') {
// $size should be number
}
```

**Review by chorobin — CHANGES_REQUESTED** [review]

File: src/components/Icon/IconStyles.tsx (hunk 1 above)
Comment (samuelg0rd0n) [reply src/components/Icon/IconStyles.tsx]: Wow, TBH, I had no idea about the concept of "type guards" 😀  Pretty cool. And actually this piece of code was used in two places so this function was exactly what it needed. Thanks.

File: src/components/Icon/IconStyles.tsx [hunk 2]
```diff
@@ -3,30 +3,37 @@ import styled from 'styled-components'
...
-		if (Spacing.includes(size as unknown as Spacing)) {
+export const StyledIconContainer = styled.span<StyledIconProps>`
+	display: inline-flex;
+
+	${({ $size, theme }): string => {
+		if (isSpacing($size)) {
 			return `
-				width: ${theme.$pc.spacing[size as unknown as Spacing]}px;
-				height: ${theme.$pc.spacing[size as unknown as Spacing]}px;
+				width: ${theme.$pc.spacing[$size as unknown as Spacing]}px;
```
Comment (chorobin) [inline src/components/Icon/IconStyles.tsx:18]: Do we still need the casts if we have the type guard?

**Review by chorobin — CHANGES_REQUESTED** [review]

File: src/components/Icon/IconStyles.tsx (hunk 2 above)
Comment (samuelg0rd0n) [reply src/components/Icon/IconStyles.tsx]: Of course we don't :-) I fixed it.

**Review by chorobin — APPROVED** [review]

**Pull request merged**